EventMixing/src/ProtonBunchIntensityLogNormal_module.cc

```diff
@@ -57,28 +61,32 @@ namespace mu2e {
   ProtonBunchIntensityLogNormal::ProtonBunchIntensityLogNormal(const Parameters& conf)
     : art::EDProducer{conf}
     , urbg_(createEngine(art::ServiceHandle<SeedService>()->getSeed()))
-    , lognd_(solveForMu(conf().extendedMean(), conf().sigma()),
-             conf().sigma())
+    , debug_(conf().debug())
     , mean_(conf().extendedMean())
     , cutMin_(conf().cutMin())
     , cutMax_(conf().cutMax())
   {
     produces<mu2e::ProtonBunchIntensity>();
     produces<mu2e::ProtonBunchIntensity,art::InSubRun>("MeanIntensity");
+    double sigma(0.0), SDF(0.0);
+    if(conf().sigma(sigma) && conf().SDF(SDF))
+      throw cet::exception("BADCONFIG")<<"Only one of 'sigma' or 'SDF' can be specified" << std::endl;
+    if((!conf().sigma(sigma)) && (!conf().SDF(SDF)))
+      throw cet::exception("BADCONFIG")<<"One of 'sigma' or 'SDF' must be specified" << std::endl;
+    if(conf().SDF(SDF)) sigma = sqrt(-log(SDF));
+    lognd_ = std::lognormal_distribution<double>(solveForMu(conf().extendedMean(),sigma),sigma);
```

Comment:
The requirement to specify only one of sigma and SDF can be expressed in FHICL: keep sigma as OptionalAtom,
but define SDF as an Atom with the use_if predicate that returns true if sigma is not specified.  See
https://cdcvs.fnal.gov/redmine/projects/fhicl-cpp/wiki/Conditional_configuration
for details.